opentimelineio/adapters/cmx_3600.py

```diff
@@ -687,123 +687,236 @@ def write_to_string(input_otio, rate=None, style='avid'):
+    if clip.media_reference:
+        if hasattr(clip.media_reference, 'target_url'):
+            url = clip.media_reference.target_url
+    else:
+        url = clip.name
+
+    if from_or_to not in ['FROM', 'TO']:
+        raise otio.exceptions.NotSupportedError(
+            "The clip FROM or TO value '{}' is not supported.".format(
+                from_or_to
+            )
+        )
 
-        lines.append("")
-        edit_number += 1
+    if clip.name:
+        # Avid Media Composer outputs two spaces before the
+        # clip name so we match that.
+        lines.append("* {from_or_to} CLIP NAME:  {name}".format(
+            from_or_to=from_or_to,
+            name=clip.name
+        ))
+    if url and style == 'avid':
+        lines.append("* {from_or_to} CLIP: {url}".format(
+            from_or_to=from_or_to,
+            url=url
+        ))
+    if url and style == 'nucoda':
+        lines.append("* {from_or_to} FILE: {url}".format(
+            from_or_to=from_or_to,
+            url=url
+        ))
+
+    cdl = clip.metadata.get('cdl')
+    if cdl:
+        asc_sop = cdl.get('asc_sop')
+        asc_sat = cdl.get('asc_sat')
+        if asc_sop:
+            lines.append(
+                "*ASC_SOP ({} {} {}) ({} {} {}) ({} {} {})".format(
+                    asc_sop['slope'][0],
+                    asc_sop['slope'][1],
+                    asc_sop['slope'][2],
+                    asc_sop['offset'][0],
+                    asc_sop['offset'][1],
+                    asc_sop['offset'][2],
+                    asc_sop['power'][0],
+                    asc_sop['power'][1],
+                    asc_sop['power'][2]
+                ))
+        if asc_sat:
+            lines.append("*ASC_SAT {}".format(
+                asc_sat
+            ))
+
+    # Output any markers on this clip
+    for marker in clip.markers:
+        timecode = otio.opentime.to_timecode(
+            marker.marked_range.start_time,
+            edl_rate
+        )
 
-    text = "\n".join(lines)
-    return text
+        color = marker.color
+        meta = marker.metadata.get("cmx_3600")
+        if not color and meta and meta.get("color"):
+            color = meta.get("color").upper()
+        comment = (marker.name or '').upper()
+        lines.append("* LOC: {} {:7} {}".format(timecode, color, comment))
+
+    # If we are carrying any unhandled CMX 3600 comments on this clip
+    # then output them blindly.
+    extra_comments = clip.metadata.get('cmx_3600', {}).get('comments', [])
+    for comment in extra_comments:
+        lines.append("* {}".format(comment))
+
+    return lines
+
+
+def lookahead_and_behind_enumerate(iterable):
+    prv = None
+    iterator = iter(iterable)
+    cur = next(iterator)
+    for nxt in iterator:
+        yield (prv, cur, nxt)
+        prv, cur = cur, nxt
+    yield (prv, cur, None)
+
+
+def reel_from_clip(clip):
+    if (
+        clip.media_reference
+        and isinstance(clip.media_reference, otio.schema.Gap)
```

Comment:
A media_reference will never be a Gap. The `clip` argument might be a Gap.

Reply:
Great catch!  This was existing logic but I've updated it here.